plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -206,11 +207,17 @@ public final class IcebergUtil
 
     private IcebergUtil() {}
 
-    public static Table loadIcebergTable(TrinoCatalog catalog, IcebergTableOperationsProvider tableOperationsProvider, ConnectorSession session, SchemaTableName table)
+    public static Table loadIcebergTable(
+            TrinoCatalog catalog,
+            IcebergTableOperationsProvider tableOperationsProvider,
+            ForwardingFileIo fileIo,
```

Comment:
I'd probably use the interface `FileIO` for the type across the board in this commit.